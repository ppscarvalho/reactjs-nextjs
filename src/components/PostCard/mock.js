export const postCardPropsMock = {
  title: 'qui est esse 2',
  body: '@olore placeat quibusdam ea quo vitae magni quis enim qui',
  id: 1,
  cover: 'https://i.pinimg.com/originals/1e/06/e1/1e06e107f0ca520aed316957b685ef5c.jpg',
};
