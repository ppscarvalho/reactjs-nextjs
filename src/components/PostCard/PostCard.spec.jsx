import { render, screen } from '@testing-library/react';

import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PosrCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /qui est esse 2/i })).toHaveAttribute(
      'src',
      'https://i.pinimg.com/originals/1e/06/e1/1e06e107f0ca520aed316957b685ef5c.jpg',
    );

    expect(screen.getByRole('heading', { name: /qui est esse 2/i })).toBeInTheDocument();

    expect(screen.getByText('@olore placeat quibusdam ea quo vitae magni quis enim qui')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
