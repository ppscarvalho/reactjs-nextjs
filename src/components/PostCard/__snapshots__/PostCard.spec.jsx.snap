// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="qui est esse 2"
    src="https://i.pinimg.com/originals/1e/06/e1/1e06e107f0ca520aed316957b685ef5c.jpg"
  />
  <div
    class="post-content"
  >
    <h1>
      qui est esse 2
       
      1
    </h1>
    <p>
      @olore placeat quibusdam ea quo vitae magni quis enim qui
    </p>
    <p>
      Alterado
    </p>
  </div>
</div>
`;
